src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -146,14 +151,16 @@ private String getUsagePreamble() {
     }
 
     /**
+     * <p>Note: uses {@link #DEFAULT_PARSER_OPTIONS}.
+     *
      * @param callerArguments The object containing the command line arguments to be populated by
      *                        this command line parser.
      */
     public CommandLineArgumentParser(final Object callerArguments) {
         this(
                 callerArguments,
                 Collections.<CommandLinePluginDescriptor<?>>emptyList(),
-                Collections.<CommandLineParserOptions>emptySet()
+                DEFAULT_PARSER_OPTIONS
```

Comment:
Since we now want to have more complex options for things like custom expansion file extensions, I think it would be cleaner to switch over to using a configuration object that is passed in by clients. Barclay can have a default implementation, with methods like "List<String> getExpansionFileExtensions()", and custom implementations be provided by the clients.

Reply:
So do you mean to use a `BarclayConf` interface (or mutable object) to pass to the parser? I think that passing the options as single elements makes clear which behaviour is changing/requested...

If we are going to that direction, I can either make another commit here or open a new PR, whatever you prefer...

Reply:
I meant that instead of the CommandLineParserOption enum or interface, Barclay should have a CommandLineParserConfig class, with methods like "List<String> getExpansionFileExtensions()" and "boolean getAppendToCollections()", with default implementations. Any customizations would be implemented by passing a config object to the constructor which is a subclass of that default implementation. Barclay would just query it; hopefully the object itself won't need any mutators.

When we just had options that were toggles, it would have been overkill, but with the need to return a custom list, this is cleaner, and will eliminate the need for type casting, etc. Reusing this PR is fine, or make a new one if its easier - either way.